alomgir-alom310/week-4/express-personal-api/server.js

```diff
@@ -0,0 +1,71 @@
+// require express and other modules
+const express = require('express');
+const app = express();
+
+// parse incoming urlencoded form data
+// and populate the req.body object
+const bodyParser = require('body-parser');
+app.use(bodyParser.urlencoded({ extended: true }));
+
+// allow cross origin requests (optional)
+// https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS
+app.use(function(req, res, next) {
+  res.header("Access-Control-Allow-Origin", "*");
+  res.header("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");
+  next();
+});
+
+/************
+ * DATABASE *
+ ************/
+
+// const db = require('./models');
+
+/**********
+ * ROUTES *
+ **********/
+
+// Serve static files from the `/public` directory:
+// i.e. `/images`, `/scripts`, `/styles`
+app.use(express.static('public'));
+
+/*
+ * HTML Endpoints
+ */
+
+app.get('/', function homepage(req, res) {
+  res.sendFile(__dirname + '/views/index.html');
+});
+
+
+/*
+ * JSON API Endpoints
+ */
+
+app.get('/api', (req, res) => {
+  // TODO: Document all your api endpoints below as a simple hardcoded JSON object.
+  // It would be seriously overkill to save any of this to your database.
+  // But you should change almost every line of this response.
+  res.json({
+    message: "Welcome to my personal api! Here's what you need to know!",
+    documentationUrl: "https://github.com/alom310/express-personal-api/README.md",
+    baseUrl: "http://YOUR-APP-NAME.herokuapp.com", // CHANGE ME
+    endpoints: [
+      {method: "GET", path: "/api", description: "Describes all available endpoints"},
+      {method: "GET", path: "/api/profile", description: "About me"}, 
+      {method: "GET", path: "/api/destinations", description: "Places I've visited."}, 
+      {method: "POST", path: "/api/destinations", description: "Places I want to go."},
```

Comment:
I like what you did here. But, remember once you post your places you would just be adding destinations that you have visited.